server/main.go

```diff
@@ -0,0 +1,280 @@
+			r.Get("/{address}/holders", getTokenHolders)
+			r.Get("/{address}/internal_transactions", getInternalTransactions)
+		})
+		r.Route("/api/transaction", func(r chi.Router) {
+			r.Get("/{hash}", getTransaction)
+		})
+
+		r.Route("/api/richlist", func(r chi.Router) {
+			r.Get("/", getRichlist)
+		})
+
+		r.Route("/", func(r chi.Router) {
+			r.Get("/totalSupply", getTotalSupply)
+			r.Get("/circulatingSupply", getCirculating)
+
+			r.Get("/*", staticHandler)
+		})
+
+		http.ListenAndServe(":8080", r)
+		return nil
+	}
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Fatal().Err(err).Msg("Run")
+	}
+
+}
+
+func getTotalSupply(w http.ResponseWriter, r *http.Request) {
+	totalSupply, _ := backendInstance.TotalSupply()
+	total := new(big.Rat).SetFrac(totalSupply, wei) // return in GO instead of wei
+	w.Write([]byte(total.FloatString(18)))
+}
+
+func getCirculating(w http.ResponseWriter, r *http.Request) {
+	circulatingSupply, _ := backendInstance.CirculatingSupply()
+	circulating := new(big.Rat).SetFrac(circulatingSupply, wei) // return in GO instead of wei
+	w.Write([]byte(circulating.FloatString(18)))
+}
+func staticHandler(w http.ResponseWriter, r *http.Request) {
+	requestPath := r.URL.Path
+	fileSystemPath := wwwRoot + r.URL.Path
+	endURIPath := strings.Split(requestPath, "/")[len(strings.Split(requestPath, "/"))-1]
+	splitPath := strings.Split(endURIPath, ".")
+	if len(splitPath) > 1 {
+		if f, err := os.Stat(fileSystemPath); err == nil && !f.IsDir() {
+			http.ServeFile(w, r, fileSystemPath)
+			return
+		}
+		http.NotFound(w, r)
+		return
+	}
+	http.ServeFile(w, r, wwwRoot+"index.html")
+}
+
+func getCurrentStats(w http.ResponseWriter, r *http.Request) {
+	writeJSON(w, http.StatusOK, backendInstance.GetStats())
+}
+
+func getRichlist(w http.ResponseWriter, r *http.Request) {
+	totalSupply, _ := backendInstance.TotalSupply()
+	skip, limit := parseSkipLimit(r)
+	circulatingSupply, _ := backendInstance.CirculatingSupply()
+	bl := &models.Richlist{
+		Rankings:          []*models.Address{},
+		TotalSupply:       new(big.Rat).SetFrac(totalSupply, wei).FloatString(18),
+		CirculatingSupply: new(big.Rat).SetFrac(circulatingSupply, wei).FloatString(18),
+	}
+	bl.Rankings = backendInstance.GetRichlist(skip, limit)
+	writeJSON(w, http.StatusOK, bl)
+}
+
+func getAddress(w http.ResponseWriter, r *http.Request) {
+	addressHash := chi.URLParam(r, "address")
+	log.Info().Str("address", addressHash).Msg("looking up address")
+	address := backendInstance.GetAddressByHash(addressHash)
+	balance, err := backendInstance.BalanceAt(addressHash, "pending")
+	if err == nil {
+		address.BalanceWei = balance.String()                                                                                      //to make sure that we are showing most recent balance even if db is outdated
+		address.Balance, _ = new(big.Float).SetPrec(100).Quo(new(big.Float).SetInt(balance), new(big.Float).SetInt(wei)).Float64() //converting to GO from wei
```

Comment:
This is not sufficient - the precision is lost when converting to float64. Can we just use `big.Rat` like above to remain *exact*, and build the string ourselves with `FloatString(18)`?